### PR TITLE
MTN Objective.to_dict > Objective.get_objective + workflows

### PR DESCRIPTION
**Pull request by Badr-MOUFAD**

### CODE REVIEW

**Comment by mathurinm** [comment]
@Badr-MOUFAD can you investigate the failure?

File: solvers/sklearn.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def set_objective(self, X, y, lmbd, fit_intercept):
         self.fit_intercept = fit_intercept
 
         n_samples = self.X.shape[0]
-        self.ridge = Ridge(alpha=self.lmbd/n_samples,
+        self.ridge = Ridge(alpha=2*self.lmbd,
```
Comment (Badr-MOUFAD) [inline solvers/sklearn.py:32]: cf. [scikit-learn doc on Ridge regression](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.Ridge.html)

Comment (tomMoral) [reply solvers/sklearn.py:31]: Can you share a non-converging curve before this change and a one converging after?

Comment (Badr-MOUFAD) [reply solvers/sklearn.py]: Screenshot from 2022-11-25 21-08-19

Comment (tomMoral) [reply solvers/sklearn.py]: I guess this is after the change? ok looks good :)

File: solvers/cd.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def cd(X, y, lmbd, L, n_iter):
                 if L[j] == 0.:
                     continue
                 old = w[j]
-                w[j] = w[j] + X[:, j] @ R / L[j] + lmbd * w[j]
+                w[j] = w[j] + (X[:, j] @ R - lmbd * w[j]) / L[j]
```
Comment (mathurinm) [inline solvers/cd.py:71]: Nice catch. I looked at the file and the lipschitz constants are wrong too, there should not be divisions by 4.

**Comment by mathurinm** [comment]
Thanks @Badr-MOUFAD for the cleanups. The readme still mentions division by n_samples, could you update it too ?

File: solvers/sklearn.py (hunk 1 above)
Comment (mathurinm) [reply solvers/sklearn.py]: Since sklearn omits both divisions by 2 (in datafit and penalty) I guess you should simply pass alpha=alpha here

**Comment by mathurinm** [comment]
I'll merge as this contains important fixes, and @Badr-MOUFAD will try to fix in another PR

**Pull request merged**